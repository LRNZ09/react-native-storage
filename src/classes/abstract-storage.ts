abstract class AbstractStorage {
	protected constructor() {
		throw new Error(
			'[@lrnz09/react-native-storage]: Storage is not instantiable.',
		)
	}
}

export { AbstractStorage }
