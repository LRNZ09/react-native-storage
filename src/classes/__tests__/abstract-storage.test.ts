import { AbstractStorage } from '../abstract-storage'

class FakeStorage extends AbstractStorage {
	static getInstance = () => new FakeStorage()
}

describe('AbstractStorage', () => {
	test('should not be instantiable', async () => {
		expect(FakeStorage.getInstance).toThrowErrorMatchingSnapshot()
	})
})
