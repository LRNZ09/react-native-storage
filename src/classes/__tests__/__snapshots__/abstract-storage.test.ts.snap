// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractStorage should not be instantiable 1`] = `"[@lrnz09/react-native-storage]: Storage is not instantiable."`;
